src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/DefaultProjectWorkspaceStateGenerator.cs

```diff
@@ -116,6 +116,13 @@ public void Dispose()
 
         private async Task UpdateWorkspaceStateAsync(Project workspaceProject, ProjectSnapshot projectSnapshot, CancellationToken cancellationToken)
         {
+            // We fire this up on a background thread so we could have been disposed already, and if so, waiting on our semaphore
+            // throws an exception, and then RPS yells at us.
```

Comment:
```suggestion
            // throws an exception.
```

Reply:
I bring my whole personality to work Ryan! 😛